File: apps/backend_go/internal/domain/sync_status.go
package domain

import "time"

/*
 * SyncStatus はデータベースの全データの最新状況を表します。
 * @responsibility: 1分足、セッション、経済指標それぞれの最新更新日時をひとまとめにし、
 * MT5からの同期がどこまで完了しているかを管理する。
 */
type SyncStatus struct {
	LastCandleAt  time.Time `json:"lastCandleAt"`  // 最終足の日時
	LastSessionAt time.Time `json:"lastSessionAt"` // 最終セッション集計の日時
	LastEventAt   time.Time `json:"lastEventAt"`   // 最新の経済指標（実績）の日時
	TotalCandles  int64     `json:"totalCandles"`  // 1分足の累計件数
	SyncHealth    string    `json:"syncHealth"`    // "Healthy" | "Stale" (24h以上更新がない場合)
}

File: apps/backend/src/domain/entities/syncStatus.ts
import { z } from '@hono/zod-openapi';

/**
 * SyncStatusSchema はMT5とデータベースの同期状況を定義するドメインスキーマです。
 * @responsibility: 最終更新日時や同期の健全性（Healthy）を保持し、データの新鮮度を可視化する。
 */
export const SyncStatusSchema = z.object({
  lastCandleAt: z.string().openapi({
    example: '2026-03-27T16:00:00Z',
  }),
  lastSessionAt: z.string().openapi({
    example: '2026-03-27T00:00:00Z',
  }),
  lastEventAt: z.string().openapi({
    example: '2026-03-27T16:00:00Z',
  }),
  totalCandles: z.number().openapi({
    example: 1250,
  }),
  syncHealth: z.string().openapi({
    example: 'Healthy',
  }),
}).openapi('SyncStatus');

export type SyncStatus = z.infer<typeof SyncStatusSchema>;
